libs/barista-components/breadcrumbs/src/breadcrumbs.ts

```diff
@@ -49,38 +98,394 @@ export const _DtBreadcrumbMixinBase = mixinColor<
+    }
   }
 
   ngAfterContentInit(): void {
     this._items.changes
       .pipe(startWith(null), takeUntil(this._destroy$))
       .subscribe(() => {
-        // We need to notify the items whether they are the last one in the list,
-        // because they use this information to determine their active state.
+        // If items change, reset everything to default
+        this._itemsWidthMap.clear();
+
+        if (this._itemsPortalsMap.size > 0) {
+          // Move all transplanted items back from the overlay to the breadcrumbs
+          // so that each item's width can be calculated correctly again
+          this._itemsPortalsMap.forEach((item) => {
+            item.detach();
+          });
+
+          this._itemsPortalsMap.clear();
+          this._overlayRef?.detach();
+          // Mark breadcrumbs dirty to correctly render the overlay toggle
+          this._changeDetectorRef.markForCheck();
+        }
+
         this._items.forEach((item, index) => {
+          item._setEllipsis(false);
+          // We need to notify the items whether they are the last one in the list,
+          // because they use this information to determine their active state.
           item._setCurrent(this._items.length - 1 === index);
+          this._itemsWidthMap.set(
+            item,
+            item._elementRef.nativeElement.getBoundingClientRect().width,
+          );
         });
       });
   }
 
   ngOnDestroy(): void {
     this._destroy$.next();
     this._destroy$.complete();
+    if (this._overlayRef) {
+      this._overlayRef.dispose();
+    }
+  }
+
+  closeOverlay(): void {
+    this._overlayRef?.detach();
+  }
+
+  _createOverlay(): void {
+    if (!this._overlayRef) {
+      const overlayConfig: OverlayConfig = {
+        positionStrategy: this._overlay
+          .position()
+          .flexibleConnectedTo(this._elementRef)
+          .setOrigin(this._elementRef)
+          .withPositions(DT_BREADCRUMBS_OVERLAY_POSITIONS),
+        panelClass: 'dt-breadcrumbs-overlay',
+      };
+      this._overlayRef = this._overlay.create(overlayConfig);
+
+      this._overlayRef
+        ?.keydownEvents()
+        .pipe(
+          filter(
+            (event: KeyboardEvent) =>
+              _readKeyCode(event) === ESCAPE && !hasModifierKey(event),
+          ),
+        )
+        .subscribe(() => {
+          this.closeOverlay();
+        });
+    }
+  }
+
+  /** Toggles the overlay */
+  _toggleOverlay(): void {
+    if (this._overlayRef?.hasAttached()) {
+      this.closeOverlay();
+      this._collapsedContainer?.setAttribute(`aria-expanded`, `false`);
+      return;
+    }
+
+    this._overlayRef?.attach(new DomPortal(this._collapsedContainer));
+    this._collapsedContainer?.setAttribute(`aria-expanded`, `true`);
+    const items = Array.from(this._items).filter((item) => item._isFocusable);
+
+    const keyManager = new ActiveDescendantKeyManager(items)
+      .withWrap()
+      .withVerticalOrientation();
+
+    // Focus handling
```

Comment:
If you like you can instead create two focusTraps one in the overlay and on the breadcrumb items that are visible and react to `endAnchorListener` and `startAnchorListener` and move the focus into the other focus trap when you reach a border. This would remove the logic of controlling the direction and getting the focusable items and so on. It's quite a big change and I am ok with this here as well, but my guess is that we are missing certain edge cases for keyboard users here.